### PR TITLE
[Quant] [Inductor] add input shape check for quantized conv binary lowering

### PR DESCRIPTION
**Pull request by jiayisunx**
Add inputs shape check for quantized conv binary lowering, since qconv2d_pointwise.binary does not yet support the case of broadcasting shape inputs.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: test/inductor/test_mkldnn_pattern_matcher.py [hunk 1]
```diff
@@ -667,6 +667,36 @@ def test_qconv2d_add_relu_cpu(self):
...
+        for bias in bias_list:
+            mod = M(bias).eval()
+            x1 = torch.randn((2, 32, 9, 9))
+            x2 = torch.randn((2, 32, 1, 1))
+
+            self._test_common(
+                mod,
+                (x1, x2),
+                3,
+                10,
```
Comment (leslie-fang-intel) [inline test/inductor/test_mkldnn_pattern_matcher.py:696]: Suggest directly check `counters["inductor"]["qconv2d_binary_matcher_count"]` is zero as https://github.com/pytorch/pytorch/blob/fcf6a76108be8e7b6db528b631a7b8ebdc7470ac/test/inductor/test_mkldnn_pattern_matcher.py#L626-L642

File: test/inductor/test_mkldnn_pattern_matcher.py [hunk 2]
```diff
@@ -667,6 +667,36 @@ def test_qconv2d_add_relu_cpu(self):
     def test_qconv2d_add_relu_int8_mixed_bf16(self):
         self._qconv2d_add_cpu_test_helper(use_relu=True, int8_mixed_bf16=True)
 
+    @skipIfNoDynamoSupport
+    @skipIfNoONEDNN
+    @skipIfRocm
+    def test_qconv2d_add_broadcast_shapes_cpu(self):
+        r"""
+        This testcase will quantize Conv2d->add pattern using broadcast shape inputs.
```
Comment (leslie-fang-intel) [inline test/inductor/test_mkldnn_pattern_matcher.py:675]: We are checking the Conv2d->Add fusion will fail in this case, right?

**Review by leslie-fang-intel — APPROVED** [review]

File: test/inductor/test_mkldnn_pattern_matcher.py (hunk 2 above)
Comment (jiayisunx) [reply test/inductor/test_mkldnn_pattern_matcher.py:675]: Yes.

Comment (jgong5) [reply test/inductor/test_mkldnn_pattern_matcher.py:675]: Then, can we clarify this in the comment too?

**Review by jgong5 — APPROVED** [review]

File: test/inductor/test_mkldnn_pattern_matcher.py (hunk 2 above)
Comment (jiayisunx) [reply test/inductor/test_mkldnn_pattern_matcher.py:675]: Done, thanks.

**Review by leslie-fang-intel — APPROVED** [review]
I remember there are some UT failed due to `binary_node_inputs[x]` may be a constant and no meta field. Please check about it, others looks good to me.

**Review by jgong5 — CHANGES_REQUESTED** [review]
UT failing seems relevant.

**Review by leslie-fang-intel — APPROVED** [review]

**Comment by leslie-fang-intel** [comment]
@eellison current https://github.com/pytorch/pytorch/blob/1cc456def3df9732d292cbb8dccc3975c3f89ed0/.github/merge_rules.yaml doesn't work and @jgong5 is helping check for the failure reason. Before that, could you help to take a look of this PR?

**Review by eellison — APPROVED** [review]

**Pull request closed without merging**